cli/azd/cmd/auth_login.go

```diff
@@ -84,12 +86,15 @@ func (lf *loginFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandO
 	local.BoolVar(
 		&lf.useDeviceCode,
 		"use-device-code",
-		// For Codespaces in VSCode Browser, interactive browser login will 404 when attempting to redirect to localhost
-		// (since azd cannot launch a localhost server when running remotely).
-		// Hence, we default to device-code. See https://github.com/Azure/azure-dev/issues/1006
-		os.Getenv("CODESPACES") == "true",
+		false,
 		"When true, log in by using a device code instead of a browser.",
 	)
+	local.BoolVar(
+		&lf.skipRemoteDetection,
+		"skip-remote-override",
+		false,
+		"When true, azd will not override login method with device code when running remote, like in Codespaces.",
+	)
```

Comment:
Instead of doing this, can we:
1. Modify `useDeviceCode` to be pointer-based value (follow stringPtr implementation in `auth_login.go`)
2. Only perform remote detection if `useDeviceCode` has not been set explicitly by they user

Reply:
Updated to use stringPtr.  Thank you